gtest/src/manager.rs

```diff
@@ -546,3 +560,44 @@ impl JournalHandler for ExtManager {
         // todo!() #714
     }
 }
+
+#[derive(Debug)]
+pub struct Mailbox{
+    mail: HashMap<MessageId, Message>
+}
+
+impl Mailbox {
+    pub fn new(message: Message) -> Mailbox {
+        let mut mailbox = Mailbox{mail: HashMap::new()};
+        mailbox.insert(message);
+        mailbox
+    }
+
+    pub fn get(&self, payload: &Payload) -> Option<&Message> {
+        for message in self.mail.values(){
+            if message.payload.eq(payload){
+                return Option::Some(message);
+            }
+        }
+        None
+    }
+
+    pub fn insert(&mut self, message: Message) -> Option<Message> {
+        self.mail.insert(message.id.clone(), message)
```

Comment:
`MessageId` is a `Copy` type as far as I remember. So no need to explicitly call `clone()`.
```suggestion
        self.mail.insert(message.id, message)
```